### PR TITLE
Add the possibility to filter out or offset layers from an Inkscape svg

### PR DESCRIPTION
**Pull request by Franck78**
Start implementing some filtering of Inkscape's layers
Add a DockItem with enabled/dX/dY widgets for each valid layer discovered in the SVG file

Some things are too much for me to fix now (like doing a good visual look for the DockItem)

Some translations to fix latter
Some GUI tweaks are missing or hard to do for me (enaml !!)
Particurlarly, when using the dX dY inputs it would be nice to have
update after 3 sec of inactivity. For now, you need to disable/enable a
layer to see any change.

fixes#126

### CODE REVIEW

**Comment by Franck78** [comment]
I can start by moving the variable 'layers' inside some existing python class if you tell me which one and remove the file 'layers.py

But before doing anything, tell me if you are ok with this and more important that every svg soft uses group as layers so ignore a 'g' or apply transform at the very moment of loading the xml is ok.

-I just discovered enalm
-I never used python
-I'm  not a git user
-I have no clue what svg was made of
-I uses xml under torture
and finally I'm not a programmer and I will never be able to use enaml and all subtle construct same as you.

Just to say that it will take months to do the way you want things done !

File: inkcut/core/svg.py [hunk 1]
```diff
@@ -665,6 +685,32 @@ def parse(self, e):
...
+    def __init__(self, e):
+        """
+        Count and register G tags that are Inkscape Layers
+        Parameters
+        ----------
+            e: Element or string
+                An lxml etree.Element or an argument to pass to etree.parse()
+        """
+        # clear previous Layers info
+        del layers [:]
```
[the remark this answers is not in the document]
Comment (Franck78) [reply inkcut/core/svg.py:699]: ok, a function inside svg.py that returns a list of layers.
and the  moving of  'class layer' inside svg.py

File: inkcut/job/models.py [hunk 2]
```diff
@@ -180,6 +180,12 @@ class Job(Model):
     plot_weedline_padding = ContainerList(
         Float(), default=[10, 10, 10, 10]).tag(config=True)
 
+
+    #  I don't understand how to declare an empty list of Layers (for the layers' settings)
+    layercbbb = ContainerList( Bool(),default=[True,True,True,True,True,True,True,True,True,True] ).tag(config=True)
```
Comment (frmdstryr) [inline inkcut/job/models.py:185]: You should just be able to use:
```python
layers = ContainerList(Layer)
```

Saving it in the config doesn't really make sense here as inkcut currently doesn't save recently run jobs (only keeps them in memory).

File: inkcut/core/svg.py [hunk 3]
```diff
@@ -61,13 +62,13 @@
...
 """
 Layers = []             # list of layers discovered
-class Layer:
-    name = ''           # displayed name
-    loaded = False      # the xml for this layer is loaded or not
-    enabled = True      # when loaded, use it or not
-    ofssetX = 0
-    offsetY = 0
-    #rotate  = 0
+class Layer(Atom):
```
Comment (frmdstryr) [inline inkcut/core/svg.py:65]: Move this class definition into the job/models.py file

File: inkcut/core/svg.py [hunk 4]
```diff
@@ -178,8 +179,14 @@ def parseTransform(self, e):
         
         if name == "translate":
             dx = float(args[0])
-            dy = len(args) == 2 and float(args[1]) or dx
+            #TODO bug or explain folllowing line !
```
[the remark this answers is not in the document]
Comment (frmdstryr) [reply inkcut/core/svg.py:227]: Can you create a separate PR for this bug?

File: inkcut/core/layers.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            -rotate the layer to facilitate rearangements
+            by inserting/modifying the transform attached to the group
+            (usefull for painting mask layers)
+            -one color / layer
+            -use mouse to move layers
+"""
+
+layers = []             # list of layers discovered
+
+class Layer:
```
[the remark this answers is not in the document]
Comment (Franck78) [reply inkcut/core/layers.py]: ok, I have moved to class Layer(Atom)
The values enabled/dx/dy are saved/read from the config file BUT
-when read, my real Layers list is not updated
-I don't understand how to put my class var inside the Job class
(more explicit on the code change committed on my repo.

**Comment by frmdstryr** [comment]
> I can start by moving the variable 'layers' inside some existing python class if you tell me which one and remove the file 'layers.py

Yes the variable layers can just be another attribute on the job class, as each job/file has it's own  set of layers depending on the file.

File: inkcut/core/svg.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Franck78) [reply inkcut/core/svg.py:227]: euh, prefer not to. Not enough git skills to undo redo etc etc.
My though is that this code was borrowed from an app where sometimes parameters could be partial or complete. When the DY is missing, it uses DX (diagonal translation)
I'm not sure if there is this kind of thing in SVG files ?
so
dx = float(args[0])
dy = float(args[1]) is just enough (for me)
Note the same syntax used again 4 lines latter

File: inkcut/job/manifest.enaml [hunk 6]
```diff
@@ -48,6 +48,9 @@ def open_document(event):
     if path:
         try:
             plugin.open_document(path, nodes=event.parameters.get('nodes'))
+            #nuclear call because I have no clue to do only refresh for my layers DockItem
+            workbench.get_plugin("inkcut.ui")._refresh_dock_items()
```
[the remark this answers is not in the document]
Comment (Franck78) [reply inkcut/job/manifest.enaml:52]: I suppose that you mean Atom Layers class. This is done.
I removed the call to _refresh_dock_items() and there is no more sync.
I guess some 'observe' somewhere.
I really don't see the big picture here. who do what, when, why, etc etc etc
What may take you 5 minutes to fix can be a day for me.
I think I gave you access to my repo if you want to rapidly move on this thing (or not).

File: inkcut/job/models.py (hunk 2 above)
Comment (Franck78) [reply inkcut/job/models.py:185]: because when launched inkcut reloads the last SVG used, it make senses to me to reload also the layer setup last used.

**Comment by Franck78** [comment]
ok, its already guetting messy to read understand what is happening here.
Sorry but
my goal was to plot (hide a layer : done)
fix the init bug (done)
my goal is not
-learn python yet
-learn enlm
-learn git

I have tried to move the list of layers as NOT a basic list built and owned by svg.py. 
JOB now calls svg.py to obatain the list
JOB passes the list back to svg.py
Unless tons of mods, SVG.PY cannot use easiliy this list as a parameter maintened in JOB
Result in ugly

I would like to add something like define where is the zero (0,0) relative the machine user in front view
on a cutter it's bottom, head right, (some time head at left)
On a flat cnc , maybe the user sees it (0,0) top left etc etc 
Or add rotate transform !

So, to move, fix yourself  the many small things you don't like (because they will take days for me and it's already 7 days for that pull request).

**Comment by aaujon** [comment]
@Franck78 I cherry picked your commit on french translation, thanks for that !

**Comment by Franck78** [comment]
ah, j'ai détruit ce repo et PR parceque c'était trop bordélique. Si tu peux attendre. Je vais refaire des PR (et comprendre la logique de GIT).

**Comment by aaujon** [comment]
@Franck78 Pas de soucis, j'ai récupéré ces modifs pour etre sûr de pas les perdre.

**Comment by Franck78** [comment]
t'as pas tord parceque quand je vois les questions à la con ou explications débiles renvoyées par 'git' j'ai envie de tout balancer:

```
fatal: la branche courante layers n'a pas de branche amont.
Pour pousser la branche courante et définir la distance comme amont, utilisez
  git push --set-upstream origin layers

```

**Pull request closed without merging**